src/tests.rs

```diff
@@ -241,131 +239,90 @@ fn get_chunk_sizes() -> Result<(), Error> {
+
+    // Get the size of each chunk
+    let chunk_size = get_chunk_size(file_size, 0);
+
+    // Read the first two chunks (0 and 1)
+    let first_chunk = decrypt_range(&data_map, &encrypted_chunks, 0, chunk_size)?;
+    let second_chunk = decrypt_range(&data_map, &encrypted_chunks, chunk_size, chunk_size)?;
+
+    // Verify each chunk size
+    assert_eq!(
+        first_chunk.len(),
+        chunk_size,
+        "First chunk has incorrect size"
+    );
+    assert_eq!(
+        second_chunk.len(),
+        chunk_size,
+        "Second chunk has incorrect size"
+    );
+
+    // Join the chunks
+    let mut combined = Vec::with_capacity(2 * chunk_size);
+    combined.extend_from_slice(&first_chunk);
+    combined.extend_from_slice(&second_chunk);
+    let combined = Bytes::from(combined);
+
+    // Verify against original data
+    let expected = original_data.slice(0..2 * chunk_size);
+    assert_eq!(combined.len(), expected.len(), "Combined length mismatch");
+    compare(expected, combined)?;
 
     Ok(())
 }
 
-fn seek(
-    bytes: Bytes,
-    data_map: &DataMap,
-    encrypted_chunks: &[EncryptedChunk],
-    pos: usize,
-    len: usize,
-) -> Result<Bytes, Error> {
-    let expected_data = bytes.slice(pos..(pos + len));
-    let info = seek_info(data_map.file_size(), pos, len);
-
-    // select a subset of chunks; the ones covering the bytes we want to read
-    let subset: Vec<_> = encrypted_chunks
-        .iter()
-        .filter(|c| c.index >= info.index_range.start && c.index <= info.index_range.end)
-        .sorted_by_key(|c| c.index)
-        .cloned()
-        .collect();
-
-    let read_data = decrypt_range(data_map, &subset, info.relative_pos, len)?;
-
-    compare(expected_data, read_data.clone())?;
-
-    Ok(read_data)
+#[test]
+fn seek_with_length_over_data_size() -> Result<(), Error> {
+    let file_size = 10_000_000;
+    let bytes = random_bytes(file_size);
+    let start_pos = 512;
+
+    // Calculate length safely
+    let remaining_bytes = file_size.saturating_sub(start_pos);
+    let len = remaining_bytes.saturating_add(1); // Try to read one more byte than available
+
+    let (data_map, encrypted_chunks) = encrypt_chunks(bytes.clone())?;
+
+    // We expect to get data from start_pos to end of file
+    let expected_data = bytes.slice(start_pos..file_size);
+
+    let read_data = decrypt_range(&data_map, &encrypted_chunks, start_pos, len)?;
+    compare(expected_data, read_data)?;
+
+    // Also verify reading beyond end returns empty
+    let read_data = decrypt_range(&data_map, &encrypted_chunks, file_size + 1, 1)?;
+    assert!(
+        read_data.is_empty(),
+        "Reading beyond end should return empty"
+    );
+
+    Ok(())
 }
 
 #[test]
 fn seek_over_chunk_limit() -> Result<(), Error> {
-    // Having first chunk being at index 1 starts at position: 4_194_304
     let start_size = 4_194_300;
-    for i in 0..27 {
+    for i in 0..5 {
```

Comment:
size has been reduced for any reason ?
if aim to cover `small sized file` purposely, maybe better make as a new testing scenario?
say make this test as a function taking a parameter of `i`, and pass down different values in ?

Reply:
It was just to reduce test time, the larger size was not testing anything further AFAIK.